forte/data/base_store.py

```diff
@@ -31,6 +32,81 @@ def __init__(self):
         various fields stored in entry lists.
         """
 
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        return state
+
+    def __setstate__(self, state):
+        self.__dict__.update(state)
+
+    def serialize(
+        self,
+        output_path: str,
+        serialize_method: str = "json",
+        save_attribute: bool = True,
+    ):
+        if serialize_method == "json":
+            with open(output_path, mode="wt", encoding="utf-8") as json_out:
+                json_out.write(self.to_string(serialize_method, save_attribute))
+        else:
+            raise NotImplementedError(
+                f"Unsupported serialization method {serialize_method}"
+            )
+
+    def to_string(
+        self,
+        json_method: str = "json",
```

Comment:
a bit confusing, what other `json_method`s are possible?

Reply:
Now it only supports JSON. It could support jsonpickle later.